### DISCUSSION TITLE
SimpleXMLElement response deprecated

### DISCUSSION THREAD

**Discussion by bnzo** [saloonphp/saloon#348] [Q&A]
Hey 👋

I am not sure to understand why returning a response as a SimpleXMLElement is deprecated. 
https://github.com/saloonphp/saloon/blob/bf4157098f4c867999e031062865265e34cbd645/src/Http/Response.php#L240

I actually do a lot of manipulations using SimpleXMLElement objects on a specific API.
I tried to use the XmlWrangler but I am not sure how to use it for specific cases like root attributes updates and also simply return the full modified xml from the reader.

Example of actual usage (simplified)

```php
public function refresh(): bool
{
    //retrieve an xml like <item attr1="foo" attr2="bar" saved="false" /> 
    $xml = $this->connector->send(new GetXmlRequest($this->document))->xml();

    //simply modify the saved attribute
    $xml['saved'] = 'false';

    //send back the modified xml
    $response = $this->connector->send(new SaveXmlRequest($this->document, $xml));

    return $response->successful();
}
```

For now I just made a custom response class (which is fine to prevent phpstan yelling at me and continue XML manipulations:

```php
class CustomResponse extends Response
{
    /**
     * Convert the XML response into a SimpleXMLElement.
     *
     * @not-deprecated
     */
    public function xml(mixed ...$arguments): SimpleXMLElement|bool
    {
        return simplexml_load_string($this->body(), ...$arguments);
    }
}
```

Maybe I am missing something on how to properly use the XMLWrangler? Any Advices?

Thanks!

**Comment by Sammyjo20** [comment]
Hey @bnzo thank you for pointing this out. Admittedly I think I might have been too quick to deprecate the method, it's just I thought it wasn't very widely used and every time I tried to use it, it didn't work. I'd be happy to un-deprecate it.

I didn't consider it to be used like this - you can't really do this super easily in XML Wrangler, it's mostly for just reading OR writing, not overwriting something you're reading.

**Comment by Sammyjo20** [comment]
Hey @bnzo just a quick update on this one - in the latest release of Saloon I have chosen to remove the deprecation warning, but I've added some additional wording to help people choose the methods. It's not the best to have `xml` and `xmlReader` but at least they have different usages.

https://github.com/saloonphp/saloon/pull/350

**Reply by bnzo** [reply]
Hey @Sammyjo20, thanks for the fix!
